queryMethods.cpp

```diff
@@ -20,49 +28,181 @@ void delay(int number_of_seconds)
+    FILE* stream = fopen(filePath, "r");
+
+
+    while (fgets(line, 1024, stream)) {
+        char* tmp = strdup(line);
+        currentGear[location] = atof(getfield(tmp, 34));
+        location++;
+        free(tmp);
+    }
+    fclose(stream);
+}
+void creatArrayOfTransmissionOilTemperatureValues() {
+
+    char line[1024];
+    int location = 0;
+    FILE* stream = fopen(filePath, "r");
+
+
+    while (fgets(line, 1024, stream)) {
+        char* tmp = strdup(line);
+        transmissionOilTemperature[location] = atof(getfield(tmp, 35));
+        location++;
+        free(tmp);
+    }
+    fclose(stream);
+}
+void creatArrayOfVehicleSpeedValues() {
+
+    char line[1024];
+    int location = 0;
+    FILE* stream = fopen(filePath, "r");
+
+
+    while (fgets(line, 1024, stream)) {
+        char* tmp = strdup(line);
+        vehicleSpeed[location] = atof(getfield(tmp, 44));
+        location++;
+        free(tmp);
+    }
+    fclose(stream);
+}
+void creatArrayOfAccelerationSpeedLongitudinalValues() {
+
+    char line[1024];
+    int location = 0;
+    FILE* stream = fopen(filePath, "r");
+
+
+    while (fgets(line, 1024, stream)) {
+        char* tmp = strdup(line);
+        accelerationSpeedLongitudinal[location] = atof(getfield(tmp, 45));
+        location++;
+        free(tmp);
+    }
+    fclose(stream);
+}
+void creatArrayOfIndicationOfBreakSwitchValues() {
+
+    char line[1024];
+    int location = 0;
+    FILE* stream = fopen(filePath, "r");
+
+
+    while (fgets(line, 1024, stream)) {
+        char* tmp = strdup(line);
+        indicationOfBreakSwitch[location] = atof(getfield(tmp, 46));
+        location++;
+        free(tmp);
+    }
+    fclose(stream);
+}
+
+int main()
+{
+    creatArrayOfFuelConsumptionValues();
+    creatArrayOfEngineSpeedValues();
+    creatArrayOfEngineCoolantTemperatureValues();
+    creatArrayOfCurrentGearValues();
+    creatArrayOfTransmissionOilTemperatureValues();
+    creatArrayOfVehicleSpeedValues();
+    creatArrayOfAccelerationSpeedLongitudinalValues();
+    creatArrayOfIndicationOfBreakSwitchValues();
+
+    for (int i = 0; i < 20; i++) {
+        printf("Main output FC %f \n", fuelConsumption[i]);
+        delay(1);
+    }
+    for (int j = 0; j < 10; j++) {
+        printf("Main output ES %f \n", engineSpeed[j]);
+        delay(1);
     }
 
+    for (int j = 0; j < 10; j++) {
+        printf("Main output EC %f \n", engineCoolantTemperature[j]);
+        delay(1);
+    }
 
+  //***** Code to get individual row and column ********//
+    //*************************************************//
+  /*  char line[1024];
+    int countVal = 0;
+    int rowNumberInd;
+    while (fgets(line, 1024, stream))
+    {        
+        char* tmp = strdup(line);
+        double desiredvalue;
+        desiredvalue = atof(getfield(tmp, 1, 1));
+        //printf("My field is col 3: %s\n", getfield(tmp,1 , 1));
+        if (countVal == rowNumberInd) {
+            printf("Hope it Works this time %f", desiredvalue);
```

Comment:
Does it work though?

Reply:
from what i tested it gave me the right values, dont need it tho, its just incase you would need it so we dont have to remake it

Reply:
Can you move these query methods into a new file.